admin/create-theme/theme-utils.php

```diff
@@ -185,4 +185,55 @@ public static function is_valid_screenshot( $file ) {
 		return 0;
 	}
 
+	public static function is_valid_screenshot_file( $file_path ) {
+		return Theme_Utils::get_screenshot_file_extension( $file_path ) !== null;
+	}
+
+	public static function get_screenshot_file_extension( $file_path ) {
+		$allowed_screenshot_types = array(
+			'png'  => 'image/png',
+			'gif'  => 'image/gif',
+			'jpg'  => 'image/jpeg',
+			'jpeg' => 'image/jpeg',
+			'webp' => 'image/webp',
+			'avif' => 'image/avif',
+		);
+		$filetype                 = wp_check_filetype( $file_path, $allowed_screenshot_types );
+		if ( in_array( $filetype['type'], $allowed_screenshot_types, true ) ) {
+			return $filetype['ext'];
+		}
+		return null;
+	}
+
+	public static function process_screenshot( $file_path ) {
+		$new_screeenshot_id      = attachment_url_to_postid( $file_path );
+		$new_screenshot_metadata = wp_get_attachment_metadata( $new_screeenshot_id );
+		$upload_dir              = wp_get_upload_dir();
+
+		$new_screenshot_location = path_join( $upload_dir['basedir'], $new_screenshot_metadata['file'] );
+
+		// TODO: We're not actually processing anything.  The image might be of the wrong dimensions, too large, etc.
+		// We're just getting the file location so that we can copy it.
+		// Should we process it?  Or just enforce that the screenshot is the correct size?
```

Comment:
Wondering if we should actually do any processing to the media selected (resize, compress, etc) or just use it as we get it and enforce sizes etc?